bin/oneoff/data_fix/remove_nonus_race_data

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env ruby
+
+#
+# Script to remove race data from users who might have been affected by the
+# geolocation bug.
+#
+
+require_relative '../../../dashboard/config/environment'
+
+# We specifically care about users that:
+# 1. Have race data and
+# 2. Have either:
+#     a. a blank geoloc (cleared out in FND-592) or
+#     b. have been geolocated to outside the US (they've been re-geolocated
+#     since their IP/geoloc was cleared out in FND-592).
+have_race = User.with_deleted.includes(:user_geos).where.not(races: nil)
+
+affected_users = have_race.
+  where(user_geos: {id: nil}).
+  or(have_race.where.not(user_geos: {country: "United States"}))
+
+affected_users.all.each do |user|
+  # we can ignore users who have not signed in since the proxies list was
+  # updated prior to https://github.com/code-dot-org/code-dot-org/pull/28635
+  old_account = user.current_sign_in_at.present? &&
+    user.current_sign_in_at < "2017-11-17 12:25:11 -0800".to_datetime
+  next if old_account
+
+  # Of the users (mostly students) with URM data but no geolocation, about 60%
+  # are in a section of a teacher in the US. So: Let's KEEP race data for
+  # students with _teachers_ whose geolocation is in the US OR with teachers
+  # who are associated with a school in the US, so that we can reduce the
+  # proportion of students with data we're wiping from 13% to 8%.
+  has_us_teacher = user.student? &&
+    user.teachers.any? do |teacher|
+      teacher.within_united_states? ||
+        teacher.user_school_infos.any? {|usi| usi.school_info.usa?}
+    end
+
+  next if has_us_teacher
+
+  user.races = nil
+  user.save!
```

Comment:
Aren't there a bunch of old users which fail validation for whatever reason? Will that cause a bunch of unnecessary erroring updates? Or is that intentional?

Reply:
oh, good point. There are, and I'm not sure what we should do about that. @dju90 @joshlory have we figured out a policy that would dictate how we should handle making updates to invalid users?

Reply:
Unfortunately, we haven't figured that out yet; there are jira items in our backlog under the "multiauth transition" epic dedicated to figuring out how to deal with these types of accounts. Wrt this work, I'll defer to you and Josh - my only request is that we keep user-felt impact to a minimum.

Reply:
I created a db clone to do some trials against; i have it right now checking for valid/invalid users and of the 50k users it's checked so far it's only found one invalid one.

We might be able to use this process for valid users, and do something more targeted for invalid users.

Reply:
Looks like that one invalid user was actually the only one in this whole set! So I'm happy moving forward with this as-is and just manually updating that one user.